library/src/main/java/com/novoda/downloadmanager/NetworkFileDownloader.java

```diff
@@ -59,7 +59,12 @@ private void processResponse(Callback callback, HttpClient.NetworkResponse respo
             }
         } else {
             Logger.e("Network response code is not ok, responseCode: " + responseCode);
-            callback.onError();
+            String networkErrorMessage = String.format(
+                    "Request: %s with response code: %s failed.",
+                    url,
+                    responseCode
+            );
```

Comment:
Same as before, concatenation may be more efficient and potentially easier to read (Shame we can use Kotlin's template, as it works as appending and reads amazing)